Celeste.Mod.mm/MonoModRules.Mod.cs

```diff
@@ -38,5 +48,17 @@ static partial class MonoModRules {
 
             MonoModRule.Flag.Set("LegacyMonoMod", isLegacyMonoMod);
         }
+
+        // in practice, people wanting to spawn extensions could use the new LevelExt method (scene as Level).Add(Entity e, EntityData d);
+        public static void PatchModSpawnEntityData(ILContext context, CustomAttribute attrib) {
+            FieldReference f_Entity_EntityData = MonoModRule.Modder.Module.GetType("Monocle.Entity").FindField("EntityData");
+            ILCursor cursor = new ILCursor(context);
+
+            cursor.GotoNext(MoveType.Before, i => i.MatchCall<Scene>("Add"));
+            cursor.Emit(OpCodes.Dup);
+            cursor.Emit(OpCodes.Ldarg_0);
+            cursor.Emit(OpCodes.Ldfld, f_Entity_EntityData);
+            cursor.Emit(OpCodes.Stfld, f_Entity_EntityData);
+        }
```

Comment:
I think we put the IL patches directly in the same file as the one they are used in now :thinking:

Reply:
ah it's used at multiple places, I see